src/rewriter/rewrite_proof_rule.cpp

```diff
@@ -125,6 +125,29 @@ void RewriteProofRule::init(ProofRewriteRule id,
   if (d_context != Node::null())
   {
     d_mt.addTerm(conc[0]);
+
+    Assert(d_context.getKind() == Kind::LAMBDA);
+    Node var = d_context[0][0];
+    Node curr = d_context[1];
+    while (curr != var)
+    {
+      size_t nchild = curr.getNumChildren();
+      size_t cfind = nchild;
+      for (size_t i = 0; i < nchild; i++)
+      {
+        if (expr::hasSubterm(curr[i], var))
```

Comment:
Note: redundant work by repeated calls to `expr::hasSubterm`. Result could be cached. Probably fine for now?

Reply:
Indeed, this is redundant. The path to the pattern variable however is expected to almost always be at depth 1. Indeed we could add a utility for finding a path, I can add a wishue for this.